components/RoadmapPage/Stage.tsx

```diff
@@ -0,0 +1,241 @@
+            }}
+          >
+            <div
+              className="text-white uppercase px-3 pb-2 md:p-6 font-[quicksand] font-[650]"
+              style={{
+                fontSize: `${(28 / 1065) * imgWidth}px`,
+              }}
+            >
+              {stageData.date ? stageData.label : "[REDACTED]"}
+            </div>
+          </div>
+          <div
+            className="absolute w-[100%] h-[100%] flex flex-col items-end justify-between
+                      left-0 top-0 z-[5] p-3 md:p-6 pointer-events-none"
+          >
+            <Image
+              src={
+                isLocked || !stageData.date
+                  ? "/assets/Roadmap/lock.svg"
+                  : "/assets/Roadmap/unlock.svg"
+              }
+              width={
+                isLocked || !stageData.date ? (36 / 1065) * slideWidth : (47.44 / 1065) * slideWidth
+              }
+              height={
+                isLocked || !stageData.date ? (47.25 / 1065) * slideWidth : (46 / 1065) * slideWidth
+              }
+              alt="not found image"
+            />
+            <div
+              className="font-[quicksand] text-white font-[700] uppercase"
+              style={{
+                fontSize: `${(28 / 1065) * imgWidth}px`,
+              }}
+            >
+              {
+                // eslint-disable-next-line no-nested-ternary
+                stageData.date
+                  ? `${
+                      months[
+                        parseInt(
+                          stageData.date.slice(
+                            stageData.date.length - 5,
+                            stageData.date.length - 2,
+                          ),
+                          10,
+                        ) - 1
+                      ]
+                    } ${
+                      stageData.certain
+                        ? parseInt(
+                            stageData.date.slice(stageData.date.length - 2, stageData.date.length),
+                            10,
+                          )
+                        : ""
+                    }`
+                  : "???????????"
+              }
+            </div>
+          </div>
+          <pre
+            className="z-[4] w-[100%] h-[100%] absolute left-0 top-0 
+            xl:pl-8 xl:pr-20 xl:py-8
+            md:px-6 md:pt-6
+            xs:px-3 xs:pt-3
+            px-3 pt-3
+            font-quicksand 
+            text-[5px] xs:text-[6.5px] md:text-[13.5px] xl:text-[19px] 
+            text-white hover:opacity-[1] opacity-0 
+            rounded-[10px] md:rounded-[20px]
+            hover:bg-gradient-to-r hover:from-[#000000cf] hover:via-[#00000080] hover:to-[#000000cf]
+            bg-transparent
+            transition duration-[200ms]"
+          >
+            <div className="drop-shadow-[0_1.2px_1.2px_rgba(0,0,0,0.8)]">{stageData.text}</div>
+          </pre>
+        </div>
+      </a>
+    </div>
```

Comment:
this component for `<Stage />` is starting to get a little big & messy, would you agree?

Maybe we can make this more readable by making some reusable `child` components? 💭

Reply:
non-blocking, just a thought